packages/land/contracts/registry/LandMetadataRegistry.sol

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+import {AccessControlEnumerableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/AccessControlEnumerableUpgradeable.sol";
+import {LandMetadataBase} from "./LandMetadataBase.sol";
+
+/**
+ * @title LandMetadataRegistry
+ * @author The Sandbox
+ * @notice Store information about the lands (premiumness and neighborhood)
+ */
+contract LandMetadataRegistry is AccessControlEnumerableUpgradeable, LandMetadataBase {
+    struct BatchSetData {
+        // baseTokenId the token id floor 32
+        uint256 baseTokenId;
+        // metadata: premiumness << 8 | neighborhoodId
+        uint256 metadata;
+    }
+
+    /// @notice This event is emitted when the metadata is set for a single land
+    /// @param tokenId the token id
+    /// @param oldNeighborhoodId the number that identifies the neighborhood before changing it
+    /// @param wasPremium true if land was premium
+    /// @param newNeighborhoodId the number that identifies the neighborhood
+    /// @param isPremium true if land is premium
+    event MetadataSet(
+        uint256 tokenId,
+        uint256 oldNeighborhoodId,
+        bool wasPremium,
+        uint256 newNeighborhoodId,
+        bool isPremium
+    );
+
+    /// @notice This event is emitted when the neighborhood name is set
+    /// @param neighborhoodId the number that identifies the neighborhood
+    /// @param name human readable name
+    event NeighborhoodNameSet(uint256 neighborhoodId, string name);
+
+    /// @notice This event is emitted when the metadata is set in batch
+    /// @param data token id and metadata
+    event BatchMetadataSet(BatchSetData[] data);
+
+    modifier onlyAdmin() {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "only admin");
+        _;
+    }
+
+    /// @dev this protects the implementation contract from behing initialized.
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice initializer method, called during deployment
+    /// @param admin_ address that have admin access and can assign roles.
+    function initialize(address admin_) external initializer {
+        __Context_init_unchained();
+        __ERC165_init_unchained();
+        __AccessControl_init_unchained();
+        __AccessControlEnumerable_init_unchained();
+        _grantRole(DEFAULT_ADMIN_ROLE, admin_);
```

Comment:
Because Ownable only let you add one address and we may want a list of addresses.
We can consider another role for `setPremium` and `setNeighborhoodId` assuming that this can hurt less than the batch operation